src/providers/cryptoauthlib/access_keys.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::Provider;
+use log::{error, warn};
+use serde::Deserialize;
+use std::fs::read_to_string;
+use std::path::Path;
+
+#[derive(Debug, Deserialize)]
+struct AccessKeyContainer {
+    access_keys: Vec<AccessKey>,
+}
+
+#[derive(Debug, Deserialize)]
+struct AccessKey {
+    slot: u8,
+    key: [u8; 32],
+}
+
+impl Provider {
+    /// Read access keys from a configuration file and setup the CALib to use them.
+    pub fn set_access_keys(
```

Comment:
Just for my understanding (I don't know much about the hardware 😅), what are access keys? How are they different from normal keys and why do they need to be put in the configuration file?

Reply:
Those keys are used for encrypting communication between CAL and a host processor so they provide another level of security. 

To understand relations between those access keys and regular keys stored in CAL you could imagine a use case of transferring (importing) a regular key to CAL when the content of that key is encrypted by an access key on the host and that encrypted content is next transferred from the host to the CAL which decrypts that encrypted key at the end.

Reply:
I see that makes sense, thanks!